internal/host/validator.go

```diff
@@ -1305,6 +1305,22 @@ func checkDomainNameResolution(c *validationContext, domainName string) Validati
 	for _, domain := range response.Resolutions {
 		if domain.DomainName != nil && *domain.DomainName == domainName {
 			if len(domain.IPV4Addresses) != 0 || len(domain.IPV6Addresses) != 0 {
+				for _,host := range c.cluster.Hosts{
```

Comment:
Why in host validation, we are iterating over all the hosts ? I think we need to check only the current host.
Also please use the already parsed inventory

Reply:
We are iterating over the cluster hosts as the intention is to check whether or not the currently resolved address is actually one of the cluster hosts. If this is the case then the aim of the ticket is to reject that host.